### ISSUE TITLE
Allow setting 'latest' as SCM tag

### ISSUE THREAD

**Issue by jetxeberria** [ansible/awx#9103]
##### ISSUE TYPE
 - Feature Idea

##### SUMMARY

Currently, and thanks to the development driven [this issue](https://github.com/ansible/awx/issues/282?_pjax=%23js-repo-pjax-container>), we can specify the tag that project must checkout to.

- If we specify nothing, it will checkout to last commit
- If we specify a tag, it will checkout to that tag

In our scenario we require a third option:
- Deployments team must use last commit: So, in project description, we set no default SCM Branch/tag
- Validations team must use certain tag: So, in template description, we ask to prompt the SCM Branch/tag
- Almost all times, that tag will be the latest one

However, this way validations team must be aware of the last tag available of the deployments project.

Feature requested: 
Allow the key 'latest' on SCM Branch/tag field, so AWX finds automatically which is the latest tag 

On git, latest tag can be retrieved with: 
```
tag=$(git describe --tags `git rev-list --tags --max-count=1`)
```

**Comment by jakemcdermott** [comment]
Can you tag your latest checkouts with a `latest` tag instead?

**Comment by jetxeberria** [comment]
> Can you tag your latest checkouts with a `latest` tag instead?

Yes, we could investigate that alternative. But it would require more complex tag management to delete previous latest and create the new latest every time a new tag is created, so I find proposal interesting to avoid that procedure.

**Comment by wenottingham** [comment]
I'm not sure we'd do this. The last tag (sequentially) may not have a meaning in most cases (especially when there are multiple branches).
